generate-catalog/src/main.rs

```diff
@@ -0,0 +1,209 @@
+use std::{fs::File, io::{BufWriter, Write}, path::PathBuf, str::FromStr};
+
+use color_eyre::eyre::{eyre, Error, OptionExt};
+use ego_tree::iter::NextSiblings;
+use lazy_static::lazy_static;
+use scraper::{ElementRef, Html, Node, Selector};
+use structopt::StructOpt;
+use regex::Regex;
+
+pub const CATALOG_URL: &'static str = "https://reveng.sourceforge.io/crc-catalogue/all.htm";
+
+#[derive(Debug, StructOpt)]
+struct Args {
+    #[structopt(short, long)]
+    url: Option<String>,
+
+    #[structopt(short, long)]
+    output: Option<PathBuf>,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    let args = Args::from_args();
+
+    let mut writer = if let Some(output) = &args.output {
+        Box::new(BufWriter::new(File::create(output)?)) as Box<dyn Write>
+    }
+    else {
+        Box::new(std::io::stdout())
```

Comment:
[nit] Might as well buffer writes to stdout to be consistent, and just box the file/stdout directly.